otherlibs/unix/execvp.c

```diff
@@ -14,7 +14,11 @@
 /**************************************************************************/
 
 #define _GNU_SOURCE  /* helps to find execvpe() */
+#ifdef __sun
+#include <strings.h>
+#else
 #include <string.h>
+#endif
```

Comment:
This makes me nervous.  Which library function is missing from `<string.h>` and is used in this file?  The solution might be to fix the use of this presumably not standard function rather than hack the includes.

Reply:
This patch is not needed for my `Solaris`/`Sun PRO C`/`SPARC` targets (both BB and my own _SunBlade 2500_ with `OpenSolaris`).

@tleedjarv could you please check if this hack can be removed? (also, please check and merge my PR)

Reply:
I found this snippet in Linux man page for `index`, so perhaps replacing `index` with `strchr` is the most appropriate solution here?
```
CONFORMING TO
       4.3BSD; marked as LEGACY in POSIX.1-2001.  POSIX.1-2008 removes the specifications of index() and rindex(), recommending strchr(3) and strrchr(3) instead.
```

Reply:
Yes, `index` seems to be an obsolete function according to https://pubs.opengroup.org/onlinepubs/009695399/functions/index.html

```
APPLICATION USAGE

The strchr() function is preferred over this function.

For maximum portability, it is recommended to replace the function call to index() as follows:

#define index(a,b) strchr((a),(b))

```

Reply:
@tleedjarv could you please try to add `-D__DEFAULT_SOURCE` to compiler arguments on `solaris/gcc` in configure in the similar way as I added `-D_XPG6` and try to build it? Does it fix the problem?

Reply:
> @tleedjarv could you please try to add `-D__DEFAULT_SOURCE` to compiler arguments on `solaris/gcc` in configure in the similar way as I added `-D_XPG6` and try to build it? Does it fix the problem?

I thought about it (I can't try at the moment) but I don't think this will work because it seems to be a glibc-only thing.
Konstantin, on your systems, is `<strings.h>` included in `<string.h>`? I wonder what is the reason this doesn't break for you.

Reply:
Thanks a lot for the investigation.  Indeed, `index()` is a BSD-ism and the proper function to use is `strchr`.  I'm guilty for this mistake, so I'll fix it directly.  After that, no `#ifdef` will be necessary, plain `#include <string.h>` should work.

Reply:
Fixed in commit 85849f602 .